src/Compiler/Interactive/fsi.fs

```diff
@@ -1329,23 +1329,34 @@ let internal mkBoundValueTypedImpl tcGlobals m moduleName name ty =
     entity, v, CheckedImplFile.CheckedImplFile(qname, [], mty, contents, false, false, StampMap.Empty, Map.empty)
 
 
-let scriptingDirectory =
-    let createDirectory directory =
+let scriptingSymbolsPath =
+    let createDirectory path =
         lazy
             try
-                if not (Directory.Exists(directory)) then
-                    Directory.CreateDirectory(directory) |> ignore
+                if not (Directory.Exists(path)) then
```

Comment:
.Exists not needed, CreateDirectory already does that check:

_Creates all directories and subdirectories in the specified path unless they already exist._